### PR TITLE
chore: Update OpenTelemetry Proto Definitions to 0.19.0

### LINKED ISSUE #989: Bump otel-proto
opentelemetry-proto points to 0.14.0
Bump to https://github.com/open-telemetry/opentelemetry-proto/releases/tag/v0.19.0

### PR DESCRIPTION
**Pull request by hdost**
Fixes #989

### CODE REVIEW

**Comment by hdost** [comment]
There are a number of breakages here as concept names did change somewhat on the protobuf. Working through the changes.

**Comment by TommyCpp** [comment]
Do we want to release it along with 0.19? Personally I feel like we can skip this on this release

**Comment by hdost** [comment]
> Do we want to release it along with 0.19? Personally I feel like we can skip this on this release

I'd agree. Don't want to block it further.

**Comment by cijothomas** [comment]
Any changelog updates required for this change as this can be user impacting. ?

**Comment by jtescher** [comment]
Think after the changelog entry we're good to merge this one

**Review by TommyCpp — COMMENTED** [review]
Some nit but overall looks good

File: opentelemetry-sdk/src/metrics/sdk_api/number.rs [hunk 1]
```diff
@@ -86,6 +86,120 @@ impl AtomicNumber {
...
+            }
+        }
+    }
+
+    /// Sets the number to the new maximum.
+    pub fn fetch_set_max(&self, number_kind: &NumberKind, val: &Number) {
+        match number_kind {
+            NumberKind::I64 => {
+                let mut old = self.0.load(Ordering::Acquire);
+                if (old as i64) >= (val.0 as i64) {
```
[the remark this answers is not in the document]
Comment (hdost) [reply opentelemetry-sdk/src/metrics/sdk_api/number.rs:151]: Yea I realized it was a bit much, but I thought partially wanted to get the first round working. 
For the shared functionality i was thinking to make it `#[inline]`

**Review by jtescher — APPROVED** [review]
@hdost any other changes you still want to make here? would help me to have this merged for the new metrics

**Comment by jtescher** [comment]
Ok merging as there aren't other concerns outside of the metrics SDK which is being updated

**Pull request merged**